### ISSUE TITLE
Inconsistent array dimensions

### ISSUE THREAD

**Issue by Berkmann18** [mljs/random-forest#13]
I'm trying to get this model to work with a [label-to-category dataset](https://github.com/all-contributors/ac-learn/blob/dev/src/dataset.json) but I get this error and the docs didn't help in seeing what was wrong.

```shell
/mnt/c/Users/max/Projects/ac-learn/node_modules/ml-matrix/matrix.js:3101                                                                                                          
  throw new RangeError('Inconsistent array dimensions');
  ^
RangeError: Inconsistent array dimensions
at new Matrix (/mnt/c/Users/max/Projects/ac-learn/node_modules/ml-matrix/matrix.js:3101:17)
at Function.checkMatrix (/mnt/c/Users/max/Projects/ac-learn/node_modules/ml-matrix/matrix.js:1373:47)
at RandomForestClassifier.train (/mnt/c/Users/max/Projects/ac-learn/node_modules/ml-random-forest/lib/index.js:154:33)
at Object.<anonymous> (/mnt/c/Users/max/Projects/ac-learn/index.js:17:12)
at Module._compile (internal/modules/cjs/loader.js:799:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:810:10)
at Module.load (internal/modules/cjs/loader.js:666:32)
at tryModuleLoad (internal/modules/cjs/loader.js:606:12)
at Function.Module._load (internal/modules/cjs/loader.js:598:3)
at Function.Module.runMain (internal/modules/cjs/loader.js:862:12)
```

Here's the code:
```js
const RFClassifier = require('ml-random-foreset').RandomForestClassifier
const LabelDataset = require('./src/dataset')

const trainingSet = LabelDataset.getLabels()
const predictions = LabelDataset.getCategories().map(elem =>
  LabelDataset.getDistinctCategories().indexOf(elem),
)

const options = {
  seed: 3,
  maxFeatures: 0.8,
  replacement: true,
  nEstimators: 25,
}

const classifier = new RFClassifier(options)
classifier.train(trainingSet, predictions)
const result = classifier.predict(trainingSet)
/* eslint-disable no-console */
console.log('result=')
console.dir(result)
/* /* eslint-enable no-console */
```

I'm fairly new to ML (especially in JS) so I can't tell if something in `options` is wrong or if I should use `ml-cart` (which gave the exact same error with the `options from the README) or if the classification model that suits my problem (classifying a set of GitHub labels into categories) is neither Random Forest or Decision Tree.

**Comment by OungKennedy** [comment]
Late reply but I had similar issues so I may be able to provide feedback.

1. Check that subarrays in the data set are all numbers and not strings
2.  Ensure the subarrays all have the same length.

**Comment by Berkmann18** [comment]
@OungKennedy I see, do you have an example?

**Comment by OungKennedy** [comment]
For my case my data was from a local txt file. 
So what I did was 
>var train_X = fs.readFileSync('machine-learning/activity/X_train.txt', 'utf-8').toString().split('\n');
>var train_y = fs.readFileSync('machine-learning/activity/y_train.txt', 'utf-8').toString().split('\n').map(x => x == 'active'? 1 : 0);   

This gives me a 2d array of _strings_ for train_X, and a 1d array of numbers for train_Y.  I had to convert the strings to numbers later on.
What gave me the `Inconsistent array dimensions` bug was that my data file had an empty line at the end.
So the raw data looked something like 
>1, 2, 3
>[empty line]

This means train_X was 
[ [1,2,3],[] ]
which would cause an error.
Hope this helps.
